lib/portlayer/exec/container.go

```diff
@@ -323,12 +323,25 @@ func (c *Container) Refresh(ctx context.Context) error {
 		return err
 	}
 
-	// sync power state (see issue 4872)
+	var started bool
+	// determine if the containerVM has started
+	if session, exists := c.ExecConfig.Sessions[c.ExecConfig.ID]; exists {
+		if session.Started == "true" {
+			started = true
+		}
+	}
+
+	// conditionally sync state (see issue 4872, 6372)
 	switch c.containerBase.Runtime.PowerState {
 	case types.VirtualMachinePowerStatePoweredOn:
-		c.state = StateRunning
+		// only set to running if the container process has started
+		if started {
```

Comment:
I think this will mean that we report "Running" for a containerVM that's been out-of-band restarted but before it's actually up.
We _could_ see if volatile guestinfo keys are dropped during a soft power cycle (IIRC they are not, but worth checking) and if so use one as an indicator of freshness.

Reply:
I've approve it as-is because it's needed for #6557 and it's a significant improvement over current state.